migrations/default/2_create_triggers/up.sql

```diff
@@ -0,0 +1,380 @@
+-- the  unit_id column based on the crash_id and unit_nbr
+--
+create or replace function db.people_cris_set_unit_id()
+returns trigger
+language plpgsql
+as $$
+DECLARE
+   unit_record record;
+BEGIN
+    SELECT INTO unit_record *
+        FROM db.units_cris where crash_id = new.crash_id and unit_nbr = new.unit_nbr;
+    new.unit_id = unit_record.id;
+    RETURN new;
+END;
+$$;
+
+create trigger set_cris_person_unit_id
+before insert on db.people_cris
+for each row
+execute procedure db.people_cris_set_unit_id();
+
+--
+-- handle a people record insert by copying
+-- to the vz_ and unified tables
+--
+create or replace function db.people_cris_insert_rows()
+returns trigger
+language plpgsql
+as
+$$
+BEGIN
+    -- insert new unified/truth record
+    INSERT INTO db.people (id, unit_id, prsn_nbr, unit_nbr, is_primary) values (
+        new.id, new.unit_id, new.prsn_nbr, new.unit_nbr, new.is_primary
+    );
+    -- insert new (editable) vz record (only record ID)
+    INSERT INTO db.people_vz (id) values (new.id);
+    RETURN NULL;
+END;
+$$;
+
+create or replace trigger insert_new_people_cris
+after insert on db.people_cris
+for each row
+execute procedure db.people_cris_insert_rows();
+
+--
+-- handle a cris crash update by updating the
+-- unified crash record from cris + vz values
+--
+create or replace function db.crashes_cris_update()
+returns trigger
+language plpgsql
+as $$
+declare
+    new_cris_jb jsonb := to_jsonb (new);
+    old_cris_jb jsonb := to_jsonb (old);
+    vz_record_jb jsonb;
+    column_name text;
+    updates_todo text [] := '{}';
+    update_stmt text := 'update db.crashes set ';
+begin
+    -- get corresponding the VZ record as jsonb
+    SELECT to_jsonb(crashes_vz) INTO vz_record_jb from db.crashes_vz where db.crashes_vz.crash_id = new.crash_id;
+
+    -- for every key in the cris json object
+    for column_name in select jsonb_object_keys(new_cris_jb) loop
+        -- if the new value doesn't match the old
+        if(new_cris_jb -> column_name <> old_cris_jb -> column_name) then
+            -- see if the vz record has a value for this field
+            if (vz_record_jb ->> column_name is  null) then
+                -- this value is not overridden by VZ
+                -- so update the unified record with this new value
+                updates_todo := updates_todo || format('%I = $1.%I', column_name, column_name);
+            end if;
+        end if;
+    end loop;
+    if(array_length(updates_todo, 1) > 0) then
+        -- complete the update statement by joining all `set` clauses together
+        update_stmt := update_stmt
+            || array_to_string(updates_todo, ',')
+            || format(' where db.crashes.crash_id = %s', new.crash_id);
+        raise notice 'Updating unified crash record from CRIS update: %', update_stmt;
+        execute (update_stmt) using new;
+    else
+        raise notice 'No changes to unified record needed';
+    end if;
+    return null;
+end;
+$$;
```

Comment:
these update functions are very similar across all tables, with one version for cris update and one version for a VZ update. as Rose mentioned in her PR, we might be able to make these reusable, but i'm not sure if it's worth the trouble?

Reply:
I think that would be cool but abstracting these further pushes us beyond the point where we can easily see the small differences in the functions and understand how CRIS updates are handled vs. VZ updates. They are going to be pretty set it and forget it too. Also, i'm just here really appreciating the way these handles the updated values. 🚀